public/app/features/explore/CorrelationEditorModeBar.tsx

```diff
@@ -0,0 +1,163 @@
+        panes.forEach((pane) => {
+          dispatch(removeCorrelationHelperData(pane[0]));
+          dispatch(runQueries({ exploreId: pane[0] }));
+        });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [correlationsEditorMode, correlationDetails?.dirty]);
+
+  return (
+    <>
+      {/* Handle navigating outside of Explore */}
+      <Prompt
+        message={(location, action) => {
+          if (location.pathname !== '/explore' && correlationsEditorMode && (correlationDetails?.dirty || false)) {
+            return 'You have unsaved correlation data. Continue?';
+          } else {
+            return true;
+          }
+        }}
+      />
+
+      {showSavePrompt && (
+        <CorrelationUnsavedChangesModal
+          onDiscard={() => {
+            // if we are discarding the in progress correlation, reset everything
+            dispatch(changeCorrelationDetails({ dirty: false }));
+          }}
+          onCancel={() => {
+            // if we are cancelling the exit, set the editor mode back to true and hide the prompt
+            dispatch(changeCorrelationEditorMode({ correlationEditorMode: true }));
+            setShowSavePrompt(false);
+          }}
+          onSave={() => {
+            dispatch(saveCurrentCorrelation(correlationDetails?.label, correlationDetails?.description));
+            dispatch(changeCorrelationDetails({ dirty: false }));
+          }}
+        />
+      )}
+      {toShow && (
+        <div className={styles.correlationEditorTop}>
+          <HorizontalGroup spacing="md" justify="flex-end">
+            <Tooltip content="Correlations editor in Explore is an experimental feature.">
+              <Icon className={styles.iconColor} name="info-circle" size="xl" />
+            </Tooltip>
+            <Button
+              variant="secondary"
+              disabled={!correlationDetails?.canSave}
+              fill="outline"
+              className={correlationDetails?.canSave ? styles.buttonColor : styles.disabledButtonColor}
+              onClick={() => {
+                dispatch(saveCurrentCorrelation(correlationDetails?.label, correlationDetails?.description));
+              }}
+            >
+              Save
+            </Button>
+            <Button
+              variant="secondary"
+              fill="outline"
+              className={styles.buttonColor}
+              icon="times"
+              onClick={() => {
+                dispatch(changeCorrelationEditorMode({ correlationEditorMode: false }));
+              }}
+              aria-label="exit correlations editor mode"
+            >
+              Exit Correlation Editor
+            </Button>
+          </HorizontalGroup>
+        </div>
+      )}
+    </>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => {
+  const contrastColor = theme.colors.getContrastText(theme.colors.primary.main);
```

Comment:
Why do we need this? The color seems to be "white" regardless to the theme 🤔

Reply:
Because `theme.colors.primary.main` are two very similar shades of dark blue for light and dark theme, but presumably this could be any color at all. This ensures we will have contrasting text no matter what the theme has defined for that color.